code/modules/paperwork/papershredder.dm

```diff
@@ -56,6 +56,11 @@
 				paperamount = max_paper
 			update_icon()
 			return
+		else if(paper_result == -1)
```

Comment:
This won't work, because `if (paper_result)` is already triggered with `-1`. The only false value a number can have is 0. Anything else is true.

And easy fix could be to change the above check to `if (paper_result > 0)`. But you could also do this:
```DM
if (paper_result)
	if (!anchored)
		...
	if (paperamount == max_paper)
		...
	if (paper_result > 0)
		paperamount += paper_result
	user.drop_from_inventory(W)
	qdel(W)
	playsound(src.loc, 'sound/items/pshred.ogg', 75, 1)
	if (paperamount > max_paper)
		...
	update_icon()
	return
```